cyclepainter.py

```diff
@@ -305,8 +372,10 @@ def __init__(self, curve=None, initial_monodromy_point=None, cut_point=None, kap
         #####################
         # The center of CyclePainter cuts
         self.cut_point = cut_point if cut_point else self._find_cut_point()
-
-        self.monodromy_point = np.complex(np.real(self.surface.base_point) + I*self.cut_point.imag())
+        
+        # The monodromy point is calculated from the given initial monodromy point, but made to have the same imaginary part as the cut point.
+        # As no.real/np.imag calls a method, this method must be called to get the real or imaginary part. 
+        self.monodromy_point = np.complex(np.real(self.surface.base_point)() + I*np.imag(self.cut_point)())
```

Comment:
I vaguely recall that I was using `cut_point.imag()` rather than `np.imag` because `cut_point` was a _Sage_ complex number, which I think used to be incompatible with numpy's complex numbers...

In any case, if it works now, `cp.cut_point.imag()` is still being use in line 104, would you mind changing that one as well?

Reply:
cut_point.imag() was originally used, and I was getting an error due to trying to add a method and an expression which I needed to resolve when setting the cut point. cut_point.imag() works fine now I have traced the error, and so I have replaced this line of code to keep the package more consistent.

Reply:
Following up on this, it seems the behaviour of this cut point method depends on how the function defining the Riemann surface is defined (e.g. the fix I put in place worked for f=y^6-1+x^2 where x,y, defined by A.<x,y> = QQ[], but failed on the Klein curve). Now using real_part and imag_part built into Sage has worked for both, so has been implemented where seen elsewhere.